modules/nf-core/sourmash/gather/main.nf

```diff
@@ -0,0 +1,44 @@
+process SOURMASH_GATHER {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::sourmash=4.5.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourmash:4.5.0--hdfd78af_0':
+        'quay.io/biocontainers/sourmash:4.5.0--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(signature)
+    path(database)
+
+    output:
+    tuple val(meta), path('*.csv')           , optional:true, emit: result
+    tuple val(meta), path('*_unassigned.sig'), optional:true, emit: unassigned
+    tuple val(meta), path('*_matches.sig')   , optional:true, emit: matches
+    path "versions.yml"                      , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: '--ksize 31'
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    result     = "${prefix}.csv"
+    matches    = "${prefix}_matches.sig"
+    unassigned = "${prefix}_unassigned.sig"
+
```

Comment:
These flags should be optional. You should make a (boolean) value channel which then inserts them into the command. [^1]

You will also need to add such comands for the `--save-prefetch*` parameters.

[^1]: Example: https://github.com/nf-core/modules/blob/174170689d08dc614316d90f15a73a3b0ba94521/modules/nf-core/fastp/main.nf#L30

Reply:
Done